pkg/artifactory/resource/configuration/resource_artifactory_property_set.go

```diff
@@ -0,0 +1,290 @@
+		//PATCH call structure has "propertySets -> propertySet (dynamic sting). Property name and predefinedValues names are also dynamic strings".
+		//Following nested map of string structs are constructed to match the usage of PATCH call with the consideration of dynamic strings.
+		//*/
+		var constructBody = map[string]map[string]map[string]interface{}{
+			"propertySets": {
+				unpackedPropertySet.Name: {
+					"visible":    unpackedPropertySet.Visible,
+					"properties": parseProperties(unpackedPropertySet.Properties),
+				},
+			},
+		}
+
+		content, err := yaml.Marshal(&constructBody)
+
+		if err != nil {
+			return diag.Errorf("failed to marshal property set during Update")
+		}
+
+		err = SendConfigurationPatch(content, m)
+		if err != nil {
+			return diag.Errorf("failed to send PATCH request to Artifactory during Update")
+		}
+
+		d.SetId(unpackedPropertySet.Name)
+		return resourcePropertySetRead(ctx, d, m)
+
+	}
+
+	var resourceLdapSettingsDelete = func(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		propertySetConfigs := &PropertySets{}
+		unpackedPropertySet := unpackPropertySet(d)
+
+		response, err := m.(*resty.Client).R().SetResult(&propertySetConfigs).Get("artifactory/api/system/configuration")
+		if err != nil {
+			return diag.Errorf("failed to retrieve data from API: /artifactory/api/system/configuration during Read")
+		}
+		if response.IsError() {
+			return diag.Errorf("got error response for API: /artifactory/api/system/configuration request during Read")
+		}
+		var matchedPropertySet = PropertySet{} // TODO: do we match before delete or just delete based on HCL list of sets?
+
+		matchedPropertySet = PropertySet{}
+		for _, iterPropertySet := range propertySetConfigs.PropertySets {
+			if iterPropertySet.Name == unpackedPropertySet.Name {
+				matchedPropertySet = iterPropertySet
+				break
+			}
+		}
+
+		var constructBody = map[string]map[string]string{
+			"propertySets": {
+				matchedPropertySet.Name: "~",
+			},
+		}
+
+		content, err := yaml.Marshal(&constructBody)
+
+		if err != nil {
+			return diag.Errorf("failed to marshal property set during Delete")
+		}
+
+		err = SendConfigurationPatch(content, m)
+		if err != nil {
+			return diag.Errorf("failed to send PATCH request to Artifactory during Delete")
+		}
+
+		return nil
+	}
+
+	return &schema.Resource{
+		UpdateContext: resourcePropertySetsUpdate,
+		CreateContext: resourcePropertySetsUpdate,
+		DeleteContext: resourceLdapSettingsDelete,
+		ReadContext:   resourcePropertySetRead,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema:      propertySetsSchema,
+		Description: "Provides an Artifactory Property Set resource. This resource configuration corresponds to 'propertySets' config block in system configuration XML (REST endpoint: artifactory/api/system/configuration).",
+	}
+}
+
+func unpackPredefinedValues(s interface{}) []PredefinedValue {
```

Comment:
Let's move this and the one below into the function scope of `ResourceArtifactoryPropertySet`.

Reply:
done